src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemBuffer, out INamedTypeSymbol? bufferType))
+                {
+                    return;
+                }
+
+                IMethodSymbol blockCopyMethod = bufferType
+                   .GetMembers("BlockCopy")
+                   .OfType<IMethodSymbol>()
+                   .FirstOrDefault();
+
+                if (blockCopyMethod is null)
+                {
+                    return;
+                }
+
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemArray, out INamedTypeSymbol? arrayType))
+                {
+                    return;
+                }
+
+                IPropertySymbol arrayLengthProperty = arrayType
+                    .GetMembers("Length")
+                    .OfType<IPropertySymbol>()
+                    .FirstOrDefault();
+
+                context.RegisterOperationAction(context =>
+                {
+                    var invocationOperation = (IInvocationOperation)context.Operation;
+
+                    if (!invocationOperation.TargetMethod.Equals(blockCopyMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = IOperationExtensions.GetArgumentsInParameterOrder(invocationOperation.Arguments);
+
+                    if (arguments.Length != 5)
+                    {
+                        return;
+                    }
+
+                    if (arguments[0].Parameter.Type.SpecialType != SpecialType.System_Array ||
+                        arguments[1].Parameter.Type.SpecialType != SpecialType.System_Int32 ||
+                        arguments[2].Parameter.Type.SpecialType != SpecialType.System_Array ||
+                        arguments[3].Parameter.Type.SpecialType != SpecialType.System_Int32 ||
+                        arguments[4].Parameter.Type.SpecialType != SpecialType.System_Int32)
+                    {
+                        return;
+                    }
```

Comment:
This looks fragile. If we hit this, then we're in trouble. That means another overload is added, in that case a diagnostic may be valid but we're returning here because we didn't find the correct overload (because we're using FirstOrDefault for finding the method by name).

Reply:
Good catch. The `GetFirstOrDefaultMemberWithParameterInfos` extension method approach will address what was intended here.